src/main/java/core/basesyntax/ReadFromFile.java

```diff
@@ -0,0 +1,58 @@
+package core.basesyntax;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.IllegalFormatException;
+import java.util.IllegalFormatFlagsException;
+import java.util.List;
+
+public class ReadFromFile implements ReadableFile {
+    private static final String COMMA_DELIMITER = ",";
+    private static final String[] FILE_HEADER = new String[]{"type", "fruit", "quantity", "date"};
+
+    public List<List<String>> getNewData(String fileName) throws IOException {
+        if (fileName == null || fileName.length() < 4
+                || !fileName.substring(fileName.length() - 4).equals(".csv")) {
```

Comment:
Better to use endsWith() method if we really need this check

Reply:
Done